src/js/media-captions-list.js

```diff
@@ -0,0 +1,130 @@
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import {
+  Window as window,
+  Document as document,
+} from './utils/server-safe-globals.js';
+import {
+  MediaUIEvents,
+  MediaUIAttributes,
+  TextTrackKinds,
+} from './constants.js';
+import { parseTextTrackStr, splitTextTracksStr } from './utils/captions.js';
+import MediaChromeListbox from './media-chrome-listbox.js';
+
+const CC_PREFIX = '!cc';
+const isCC = (extendedTextTrackStr = '') =>
+  extendedTextTrackStr?.startsWith?.(CC_PREFIX);
+const withoutCC = (extendedTextTrackStr = '') => {
+  if (!isCC(extendedTextTrackStr)) return extendedTextTrackStr;
+  return extendedTextTrackStr.slice(CC_PREFIX.length + 1);
+};
+
+const withCC = (textTrackStr = '') => `${CC_PREFIX}:${textTrackStr}`;
+
+const DEFAULT_UNKNOWN_LABEL = 'UNKNOWN';
+const DEFAULT_CC_LABEL_PART = '(cc)';
+const DEFAULT_FORMATTER = (value) => {
+  const textTrackStr = withoutCC(value);
+  const baseLabel =
+    parseTextTrackStr(textTrackStr)?.label ?? DEFAULT_UNKNOWN_LABEL;
+  const suffix = isCC(value) ? ` ${DEFAULT_CC_LABEL_PART}` : '';
+  return `${baseLabel}${suffix}`;
+};
+const parseAsSubCCStr = (str, kind = TextTrackKinds.SUBTITLES) => {
+  if (!str) return [];
+  if (kind === TextTrackKinds.SUBTITLES) return splitTextTracksStr(str);
+  return splitTextTracksStr(str).map(withCC);
+};
+
+class MediaCaptionsList extends MediaChromeListbox {
+  static get observedAttributes() {
+    return [
+      ...super.observedAttributes,
+      MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+      MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
+      MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+      MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
+    ];
+  }
+
+  constructor({ format = DEFAULT_FORMATTER } = {}) {
+    super({ format });
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (
+      [
+        MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+        MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+      ].includes(attrName) &&
+      oldValue !== newValue
```

Comment:
Unfortunately, `attributeChangedCallback` is invoked whenever an attribute is set, even when the attribute has not, in fact changed. E.g. this will cause `attributeChangedCallback` to get invoked:

```js
const mclEl = document.querySelector('media-captions-list');
mclEl.setAttribute('media-subtitles-list', mclEl.getAttribute('media-subtitles-list'));
```

Since we decided (while implementing the "change" events) to not make media controller only propagate state if that state has, in fact, changed, this is a realistic possibility in our current architecture.